--- a/makeicecat	2020-04-28 02:06:48.912560052 +0200
+++ b/makeicecat	2020-04-28 02:07:40.172560384 +0200
@@ -193,8 +193,8 @@
 for file in $(find |grep searchplugins/list.txt)
 do
   /bin/sed /^$/d -i $file
-  sort -u $file > /tmp/sorttmp
-  cp /tmp/sorttmp $file
+  sort -u $file > ./sorttmp
+  cp ./sorttmp $file
   echo >> $file
 done
 
@@ -524,14 +524,14 @@
 find . -executable -exec chmod 750 {} \;
 find . ! -executable -exec chmod 640 {} \;
 
-find . -type f | sed -e 's/^\.\///' | sort | xargs -i echo "{}={}" > /tmp/filelist.txt
-find . -type l | sed -e 's/^\.\///' | sort | xargs -i echo "{}={}" >> /tmp/filelist.txt
+find . -type f | sed -e 's/^\.\///' | sort | xargs -i echo "{}={}" > ./filelist.txt
+find . -type l | sed -e 's/^\.\///' | sort | xargs -i echo "{}={}" >> ./filelist.txt
 
-mkisofs -D -V "\$6" -no-pad -R -apple -o /tmp/icecat-uncompressed.dmg -path-list /tmp/filelist.txt -graft-points -gid 20 -dir-mode 0750 -new-dir-mode 0750
+mkisofs -D -V "\$6" -no-pad -R -apple -o ./icecat-uncompressed.dmg -path-list ./filelist.txt -graft-points -gid 20 -dir-mode 0750 -new-dir-mode 0750
 
 
-dmg dmg /tmp/icecat-uncompressed.dmg \$4
-rm /tmp/icecat-uncompressed.dmg
+dmg dmg ./icecat-uncompressed.dmg \$4
+rm ./icecat-uncompressed.dmg
 
 find . -executable -exec chmod 755 {} \;
 find . ! -executable -exec chmod 644 {} \;
