diff --git a/software/tulip/CMakeLists.txt b/software/tulip/CMakeLists.txt
index 1f5b1a7c7..a6ad3acb6 100644
--- a/software/tulip/CMakeLists.txt
+++ b/software/tulip/CMakeLists.txt
@@ -124,13 +124,6 @@ IF(LINUX)
             RENAME application-x-tulip.png
             DESTINATION ${DESKTOP_CONFIG_INSTALL_PREFIX}/share/icons)
 
-
-    INSTALL(CODE
-      "EXECUTE_PROCESS(COMMAND update-desktop-database ${DESKTOP_CONFIG_INSTALL_PREFIX}/share/applications
-                       COMMAND update-mime-database ${DESKTOP_CONFIG_INSTALL_PREFIX}/share/mime
-                       OUTPUT_QUIET ERROR_QUIET)"
-    )
-
   ENDIF(TULIP_LINUX_DESKTOP_REGISTRATION)
 
 ENDIF(LINUX)
