diff --git a/src/utils/config.go b/src/utils/config.go
index 77118e3..ea1d9a5 100644
--- a/src/utils/config.go
+++ b/src/utils/config.go
@@ -294,6 +294,7 @@ func linuxApp() string {
 
 	potentialList := []string{
 		"/opt/spotify/",
+ 		"/opt/spotify/spotify-client/",
 		"/usr/share/spotify/",
 		"/usr/libexec/spotify/",
 		"/var/lib/flatpak/app/com.spotify.Client/x86_64/stable/active/files/extra/share/spotify/",
