--- a/setup.cfg	2016-12-17 03:51:53.000000000 +0200
+++ b/setup.cfg	2021-05-21 18:12:08.900317387 +0300
@@ -9,37 +9,31 @@
 exclude = tests/*,*/migrations/*,*/south_migrations/*
 
 [pytest]
-norecursedirs = 
-	.git
-	.tox
-	.env
-	dist
-	build
-	south_migrations
-	migrations
-python_files = 
-	test_*.py
-	*_test.py
-	tests.py
-addopts = 
-	-rxEfs
-	--strict
-	--ignore=docs/conf.py
-	--ignore=setup.py
-	--ignore=ci
-	--doctest-modules
-	--doctest-glob=\*.rst
-	--tb=short
+norecursedirs =
+    .git
+    .tox
+    .env
+    dist
+    build
+    south_migrations
+    migrations
+python_files =
+    test_*.py
+    *_test.py
+    tests.py
+addopts =
+    -rxEfs
+    --strict
+    --ignore=docs/conf.py
+    --ignore=setup.py
+    --ignore=ci
+    --doctest-modules
+    --doctest-glob=\*.rst
+    --tb=short
 
 [isort]
-force_single_line = True
-line_length = 120
-known_first_party = sphinx_py3doc_enhanced_theme
-default_section = THIRDPARTY
-forced_separate = test_sphinx_py3doc_enhanced_theme
-
-[egg_info]
-tag_build = 
-tag_date = 0
-tag_svn_revision = 0
-
+force_single_line=True
+line_length=120
+known_first_party=sphinx_py3doc_enhanced_theme
+default_section=THIRDPARTY
+forced_separate=test_sphinx_py3doc_enhanced_theme
\ No newline at end of file
